### PR TITLE
Improve ClickHouse type mapping documentation

### PR DESCRIPTION
**Pull request by tangjiangling**
## Description

> Is this change a fix, improvement, new feature, refactoring, or other?

Improvement.

> Is this a change to the core query engine, a connector, client library, or the SPI interfaces? (be specific)

ClickHouse type mapping doc.

> How would you describe this change to a non-technical end user or system administrator?

N/A.

## Related issues, pull requests, and links

Related #11448

## Documentation

(x) Sufficient documentation is included in this PR.

## Release notes

(x) No release notes entries required.

### CODE REVIEW

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 1]
```diff
@@ -162,32 +162,141 @@ in create table statement. ``ReplicatedMergeTree`` engine is not yet supported.
...
+map the same way in both directions between Trino and the data source. Refer to
+the following sections for type mapping in each direction.
+
+ClickHouse type to Trino type mapping
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector maps ClickHouse types to the corresponding Trino types following this table:
+
+.. list-table:: ClickHouse type to Trino type mapping
+  :widths: 30, 20, 50
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/clickhouse.rst:176]: `25, 25, 50` looks better considering `DECIMAL(20,0)` in Trino.

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 2]
```diff
@@ -162,32 +162,141 @@ in create table statement. ``ReplicatedMergeTree`` engine is not yet supported.
...
+  :header-rows: 1
+
+  * - ClickHouse database type
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/clickhouse.rst:179]: nit: Just "ClickHouse type" is enough to me.

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 3]
```diff
@@ -162,32 +162,141 @@ in create table statement. ``ReplicatedMergeTree`` engine is not yet supported.
...
+    - ``DECIMAL(20,0)``
+    -
+  * - ``Float32``
+    - ``REAL``
+    - ``FLOAT`` is an alias of ``Float32``
+  * - ``Float64``
+    - ``DOUBLE``
+    - ``DOUBLE`` is an alias of ``Float64``
+  * - ``Decimal``
+    - ``DECIMAL``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/clickhouse.rst:213]: Can we add `(p, s)` to both  `Decimal` & `DECIMAL`?

File: docs/src/main/sphinx/connector/clickhouse.rst [hunk 4]
```diff
@@ -162,32 +162,141 @@ in create table statement. ``ReplicatedMergeTree`` engine is not yet supported.
...
+    - ``Int64``
+    -
+  * - ``REAL``
+    - ``Float32``
+    -
+  * - ``DOUBLE``
+    - ``Float64``
+    -
+  * - ``DECIMAL``
+    - ``DECIMAL``
```
Comment (ebyhr) [inline docs/src/main/sphinx/connector/clickhouse.rst:280]: Change to `Decimal` for consistency and it would be better to add `(p, s)` likes read mapping.

**Comment by tangjiangling** [comment]
Superseded by #13578

**Pull request closed without merging**